source/extensions/filters/http/lua/lua_filter.cc

```diff
@@ -372,7 +378,6 @@ int StreamHandleWrapper::luaHttpCall(lua_State* state) {
     options.is_async_request_ = lua_toboolean(state, AsyncFlagIndex);
   }
 
-  options.request_options_.setParentSpan(callbacks_.activeSpan());
```

Comment:
I don't strictly need to move this line up higher, but it is semantically related to the `setSampled` call, which does need to sit at the top of the method body, so that it can be overridden by the `trace_sampled` option.

Nothing else in here can call `setParentSpan`.